scripts/dts/gen_defines.py

```diff
@@ -277,669 +172,97 @@ def relativize(path):
-    # 'prop'.
-
-    # Skip #size-cell and other property starting with #. Also skip mapping
-    # properties like 'gpio-map'.
-    if prop.name[0] == "#" or prop.name.endswith("-map"):
-        return False
-
-    # See write_clocks()
-    if prop.name == "clocks":
-        return False
-
-    # For these, Property.val becomes an edtlib.Node, a list of edtlib.Nodes,
-    # or None. Nothing is generated for them at the moment.
-    if prop.type in {"phandle", "phandles", "path", "compound"}:
-        return False
-
-    # Skip properties that we handle elsewhere
-    if prop.name in {
-        "reg", "compatible", "status", "interrupts",
-        "interrupt-controller", "gpio-controller"
-    }:
-        return False
-
-    return True
-
-
-def prop2value(prop):
-    # Get the single macro expansion for prop.value.
-
-    if prop.type == "boolean":
-        return 1 if prop.val else 0
-
-    if prop.type == "string":
-        return quote_str(prop.val)
-
-    if prop.type == "int":
-        return prop.val
-
-    err(f"prop2value: unsupported type {prop.type}")
+def write_idents_and_existence(node):
+    # Writes macros related to the node's identifiers
 
+    idents = []
 
-def prop2values(prop):
-    # For array properties, get a list containing the individual
-    # elements converted to C scalar constants.
-
-    if prop.type == "array":
-        return prop.val
-
-    if prop.type == "string-array":
-        return [quote_str(elm) for elm in prop.val]
-
-    if prop.type == "uint8-array":
-        return [f"0x{elm:02x}" for elm in prop.val]
-
-    err(f"prop2values: unsupported type {prop.type}")
-
-
-def prop2init(prop):
-    # For array properties, get a C array initializer for the
-    # property value.
-
-    return "{" + ", ".join(str(val) for val in prop2values(prop)) + "}"
-
-
-def write_bus(node):
-    # Generate bus-related #defines
-
-    if not node.bus_node:
-        return
-
-    if node.bus_node.label is None:
-        err(f"missing 'label' property on bus node {node.bus_node!r}")
-
-    primary_macro = None
-    def write_bus_node_for_ident(node, ident):
-        # #define DT_<IDENT>_BUS_NAME <BUS-LABEL>
-
-        nonlocal primary_macro
-        if primary_macro is None:
-            primary_macro = out(f"{ident}_BUS_NAME",
-                                quote_str(str2ident(node.bus_node.label)))
-        else:
-            out(f"{ident}_BUS_NAME", primary_macro)
-
-    for_each_ident(node, write_bus_node_for_ident)
-
-    for compat in node.compats:
-        # #define DT_<COMPAT>_BUS_<BUS-TYPE> 1
-        out(f"{str2ident(compat)}_BUS_{str2ident(node.on_bus)}", 1)
-
-
-def write_existence_flags(node):
-    # Generate #defines of the form
-    #
-    #   #define DT_INST_<instance no.>_<compatible string> 1
-    #
-    # for enabled nodes. These are flags for which devices exist.
-
+    # Aliases
+    idents.extend(f"N_ALIAS_{str2ident(alias)}" for alias in node.aliases)
```

Comment:
Could get rid of the `idents = []` assignment and do this:

```python3
idents = [f"N_ALIAS_{str2ident(alias)}" for alias in node.aliases]
```

Reply:
That's nicer.